lib/FusionInventory/Agent/Tools/Batteries.pm

```diff
@@ -43,6 +45,42 @@ sub sanitizeBatterySerial {
     return hex2dec($serial);
 }
 
+sub getCanonicalVoltage {
+    my ($value) = @_;
+
+    return unless $value;
+
+    # We expect to return voltage in mV
+    if ($value =~ /^(\d+) mV$/i) {
+        return int($1);
+    } elsif ($value =~ /^(\d+) V$/i) {
+        return int($1) * 1000;
+    }
+
+    return $value;
+}
+
+sub getCanonicalCapacity {
+    my ($value, $voltage) = @_;
+
+    return unless $value;
+
+    # We expect to return capacity in mWh, $voltage is expected to be in mV
+    if ($value =~ /^(\d+) mWh$/i) {
+        return int($1);
+    } elsif ($value =~ /^(\d+) Wh$/i) {
+        return int($1) * 1000;
+    } elsif ($value =~ /^(\d+) mAh$/i) {
+        return unless $voltage;
+        return int($1 * $voltage / 1000);
```

Comment:
Why _* voltage_ ?

Reply:
Voltage is necessary when you have capacity in mAh or Ah.
`* $voltage` : this is how a capacity should be converted from `Ah` to `Wh`